src/main/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiAction.java

```diff
@@ -0,0 +1,265 @@
+                config.dynamic.auth_failure_listeners.getListeners()
+                    .put(
+                        listenerName,
+                        new ConfigV7.AuthFailureListener(
+                            authFailureListener.get(TYPE_JSON_PROPERTY).asString(),
+                            Optional.ofNullable(authenticationBackend),
+                            ignoreHosts,
+                            authFailureListener.get(ALLOWED_TRIES_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(TIME_WINDOW_SECONDS_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(BLOCK_EXPIRY_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(MAX_BLOCKED_CLIENTS_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(MAX_TRACKED_CLIENTS_JSON_PROPERTY).asInt()
+                        )
+                    );
+                saveOrUpdateConfiguration(client, configuration, new OnSucessActionListener<>(channel) {
+                    @Override
+                    public void onResponse(IndexResponse indexResponse) {
+
+                        ok(channel, authFailureContent(config));
+                    }
+                });
+            }).error((status, toXContent) -> response(channel, status, toXContent)));
+
+    }
+
+    private ValidationResult<SecurityJsonNode> validateAuthFailureListener(SecurityJsonNode authFailureListener, String name) {
+        if (name == null) {
+            return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("name is required"));
+        }
+        if (authFailureListener.get(TYPE_JSON_PROPERTY).isNull()) {
+            return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("type is required"));
+        }
+        if (!(authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(IP_TYPE)
+            || authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(USERNAME_TYPE))) {
+            return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("type must be username or ip"));
+        }
+        if (authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(USERNAME_TYPE)
+            && (authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).isNull()
+                || !authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).asString().equals("internal"))) {
+            return ValidationResult.error(
+                RestStatus.BAD_REQUEST,
+                badRequestMessage("ip auth failure listeners must have 'internal' authentication backend")
+            );
+        }
+        if (authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(IP_TYPE)
+            && !authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).isNull()) {
+            return ValidationResult.error(
+                RestStatus.BAD_REQUEST,
+                badRequestMessage("username auth failure listeners should not have an authentication backend")
```

Comment:
I think this is wrong. IP Auth failure listener should not have auth backend.

I'm actually a little surprised that authentication_backend is a required field at all. 

Relevant section of code: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java#L403-L414

Reply:
Yea I originally had the logic reversed, but never changed the message. I fixed it in: [2a8c237](https://github.com/opensearch-project/security/pull/4641/commits/2a8c237f25239167cf854bc50143f9f7f85a4af9)